test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs

```diff
@@ -20,6 +21,63 @@ public void OpeningWorkspaceSetsPosition()
         }
 
         [Test]
+        public void VerifyRegisteredHomeWorkspace()
+        {
+            // This test verifies the HomeWorkspace was registered with the EvaluationCompleted
+            // and RefreshCompleted events upon deserialization in both xml and json
+
+            // Load an xml test file
+            string dynFilePath = Path.Combine(Dynamo.UnitTestBase.TestDirectory, @"core\serialization\serialization.dyn");
+            string testPath = Path.GetFullPath(dynFilePath);
+            ViewModel.OpenCommand.Execute(testPath);
+
+            // Verify homeWorkspace is not null
+            var homeWorkspace = ViewModel.Model.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.NotNull(homeWorkspace);
+            
+            // Verify events handlers have been registered
+            bool openXmlFired = false;
+            ViewModel.Model.EvaluationCompleted += (sender, e) => openXmlFired = true;
+            RunCurrentModel();
+            Assert.IsTrue(openXmlFired);
+
+            //Unsubscribe
+            ViewModel.Model.EvaluationCompleted -= ViewModel.Model.OnEvaluationCompleted;
+
+            // Save to json in temp location
+            string tempPath = Path.Combine(Dynamo.UnitTestBase.TestDirectory, @"core\serialization\serialization_temp.dyn");
+            ViewModel.SaveAsCommand.Execute(tempPath);
+            
+            // Close workspace
+            Assert.IsTrue(ViewModel.CloseHomeWorkspaceCommand.CanExecute(null));
+            ViewModel.CloseHomeWorkspaceCommand.Execute(null);
+            
+            // Open json temp file
+            testPath = Path.GetFullPath(tempPath);
+            ViewModel.OpenCommand.Execute(testPath);
+
+            // Verify homeWorkspace is not null
+            homeWorkspace = ViewModel.Model.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.NotNull(homeWorkspace);
+
+            // Verify events handlers have been registered
+            bool openJsonFired = false;
+            ViewModel.Model.EvaluationCompleted += (sender, e) => openJsonFired = true;
+            RunCurrentModel();
+            Assert.IsTrue(openJsonFired);
+
+            // Close workspace
+            Assert.IsTrue(ViewModel.CloseHomeWorkspaceCommand.CanExecute(null));
+            ViewModel.CloseHomeWorkspaceCommand.Execute(null);
+
+            // Delete temp file
+            File.Delete(tempPath);
+
+            //Unsubscribe
+            ViewModel.Model.EvaluationCompleted -= ViewModel.Model.OnEvaluationCompleted;
```

Comment:
this is the wrong this to unsubscribe, you need to unsubscribe your lambda functions above, to unsubscribe these you'll most likely need to name them or assign them to a variable.